iceoryx_utils/source/posix_wrapper/access_control.cpp

```diff
@@ -34,12 +33,18 @@ bool AccessController::writePermissionsToFile(const int32_t f_fileDescriptor) co
         return false;
     }
 
-    smartAclPointer_t workingACL = createACL((m_permissions.size() + m_useACLMask) ? 1 : 0);
+    auto maybeWorkingACL = createACL((m_permissions.size() + m_useACLMask) ? 1 : 0);
+
+    if (maybeWorkingACL.has_error())
+    {
+        std::cerr << "Error: Creating ACL failed." << std::endl;
+        return false;
+    }
 
     // add acl entries
     for (const auto& entry : m_permissions)
     {
-        if (!createACLEntry(workingACL.get(), entry))
+        if (!createACLEntry(maybeWorkingACL.value().get(), entry))
```

Comment:
please add a `auto& workingACL = maybeWorkingACL.value();` after the `if (maybeWorkingACL.has_error())` statement and use that instead of always calling `maybeWorkingACL.value();`

Reply:
That's a valid point, changed.